### PR TITLE
Add Band Recycling

### PR DESCRIPTION
**Pull request by Bloo-dev**
This PR aims to add a mechanic to Metallurgy that allows to use old obsidian casts as a source of metal for molds.

Obsidian Casts may be used as a source of metal: 
- Aluminium Band: `12` units
- Thorium Band: `12` units 
- Copper Band: `12` units 
- Barium Band: `12` units
- Barinium Band: `9` units Al, `3` units Ba
- Thorium Brass: `9` units Th, `3` units Cu

These values are subject to change.

This PR is breaking as it changes the NBT structure of ores, obsidian casts and tools:

- Ores: `gm4_metallurgy:{item:"ore",ore_type:"<id>"}` -> `gm4_metallurgy:{item:"ore",metal:{type:"<id>",amount:[1s],castable:1b}}`
- Casts: `gm4_metallurgy:{item:"obsidian_cast",ore_type:"<id>"}` -> `gm4_metallurgy:{item:"obsidian_cast",metal:{type:"<id>",amount:[<primary metal amount>s,<secondary metal amount>s],castable:1b}}`
- Tools: `gm4_metallurgy:{ore_type:"<id>"}` -> `gm4_metallurgy:{metal:{type:"<id>"}}`
> Note that the NBT changes mentioned above are not comprehensive. I omitted tags that were not touched by the changes.

Due to the breaking nature of these changes a fix function for obsidian casts will have to be implemented. Ores may already be fixed by their existing fix function that was modified by this PR.
- [x]  Check whether ores are automatically fixed (Works! Ores are fixed when dropping them and/or placing them down)
- [x]  Add a fix function for casts. This function should convert old casts to the new format upon being smooshed or upon being dropped onto a sand mold.

### CODE REVIEW

**Comment by hozz8528** [comment]
I tested removing old shamirs and different things happened (they still work when applied on the item though). When I tried removing an old Defuse, the obsidian simply vanished, but when I tried removing an old Helious shamir, I got a Slightly Damaged Obsidian (the same I used to apply it).

**Comment by Bloo-dev** [comment]
@misode @SpecialBuilder32 This PR is now ready for review

File: gm4_metallurgy/data/gm4_metallurgy/functions/repair_skulls.mcfunction [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
-
+# run from item
+
+# repair ores
+# copper ore
+execute if entity @s[type=item,nbt={Item:{tag:{SkullOwner:{Properties:{textures:[{Value:"eyJ0aW1lc3RhbXAiOjE0ODE0MTA4Mjk2OTAsInByb2ZpbGVJZCI6Ijk4NWIyNzVlYmI1YTQzNDRiNDM3Njg5NTI4NjNhNjNmIiwicHJvZmlsZU5hbWUiOiJTcGFya3MiLCJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvNDNjMWE3ZGEyZWJkZGI0YTk3YzI3NDA0N2RlNzFkN2RjZWJjMTc2YTBlNjE3MzBmZmQzMzIzODUzNTNlIn0sIkNBUEUiOnsidXJsIjoiaHR0cDovL3RleHR1cmVzLm1pbmVjcmFmdC5uZXQvdGV4dHVyZS81YzNjYTdlZTJhNDk4ZjFiNWQyNThkNWZhOTI3ZTYzZTQzMzE0M2FkZDU1MzhjZjYzYjZhOWI3OGFlNzM1In19fQ=="}]}}}}}] run data merge entity @s {Item:{tag:{gm4_metallurgy:{item:"ore",metal:{type:"copper",castable:1b}},CustomModelData:18,display:{Name:'{"translate":"%1$s%3427655$s","with":["Malachite Lump",{"translate":"item.gm4.metallurgy.malachite_lump"}],"italic":false,"color":"white"}',Lore:['[{"translate":"%1$s%3427655$s","with":["Contains ",{"translate":"text.gm4.metallurgy.contains"}],"italic":false,"color":"gray"},{"translate":"%1$s%3427655$s","with":["Copper",{"translate":"text.gm4.metallurgy.copper"}],"italic":false,"color":"gold"}]']}}}}
```
Comment (BluePsychoRanger) [inline gm4_metallurgy/data/gm4_metallurgy/functions/repair_skulls.mcfunction:6]: are these fix functions supposed to have the amount in them? Right now all I'm seeing is `metal:{type:"copper",castable:1b}`

Comment (Bloo-dev) [reply gm4_metallurgy/data/gm4_metallurgy/functions/repair_skulls.mcfunction]: Yup they should! Good spot, thanks!

**Pull request merged**